gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -473,13 +482,113 @@ private KeyStore getKeystore(Path keystorePath, String keystoreType, String alia
 
     readLock.lock();
     try {
-      return getKeystore(keystoreFile, keystoreType, getKeystorePassword(alias));
+      return loadKeyStore(keystorePath, keystoreType, getKeyStorePassword(alias));
     } finally {
       readLock.unlock();
     }
   }
 
-  private char[] getKeystorePassword(String alias) throws KeystoreServiceException {
+  private boolean isKeyStoreAvailable(final Path keyStoreFilePath, String storeType, char[] password) throws KeyStoreException, IOException {
+    if (keyStoreFilePath.toFile().exists()) {
```

Comment:
Can we operate on the Path directly? `Files.exists(keystorePath)`?

Reply:
There are other examples of this throughout

Reply:
All usages of `java.io.File` will be removed from this class.

Reply:
Saw you pushed a commit addressing `File` but didn't touch this file. Assuming you were waiting on my feedback on the try/catch below.